File: duolingo_rum/Models/Achievement.cs
using System;
using System.Collections.Generic;

namespace duolingo_rum.Models;

public partial class Achievement
{
    public int Id { get; set; }

    public string Code { get; set; } = null!;

    public string Name { get; set; } = null!;

    public string? Description { get; set; }

    public string? IconEmoji { get; set; }

    public int? XpReward { get; set; }

    public string? ConditionType { get; set; }

    public int? ConditionValue { get; set; }

    public virtual ICollection<UserAchievement> UserAchievements { get; set; } = new List<UserAchievement>();
}

File: android-app/app/src/main/java/com/linguaai/app/models/Achievement.kt
package com.linguaai.app.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Achievement(
    val id: Int = 0,
    val code: String = "",
    val name: String = "",
    val description: String? = null,
    @SerialName("icon_emoji")
    val iconEmoji: String? = null,
    @SerialName("xp_reward")
    val xpReward: Int? = 0,
    @SerialName("condition_type")
    val conditionType: String? = null,
    @SerialName("condition_value")
    val conditionValue: Int? = null
)
